modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/typed/value/AbstractTypedValueTestCase.java

```diff
@@ -49,9 +52,11 @@ void runAndAssertTypedValue(String flowName, Object payloadValue, MediaType medi
   void assertTypedValue(TypedValue typedValue, Object payloadValue, MediaType mediaType, Charset charset) {
     assertThat(typedValue, is(instanceOf(TypedValue.class)));
     Object value = typedValue.getValue();
-    assertThat(value, is(instanceOf(payloadValue.getClass())));
     assertThat(value, is(payloadValue));
-    assertThat(typedValue.getDataType(), is(like(payloadValue.getClass(), mediaType, charset)));
+    if (value != null) {
```

Comment:
yes, value can be null, but with this assertion we are not testing for expected null vs expected value